src/app/frontend/common/auth/service.js

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 The Kubernetes Dashboard Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {stateName as loginState} from 'login/state';
+
+/**
+ * @final
+ */
+export class AuthService {
+  /**
+   * @ngInject
+   */
+  constructor(
+      $cookies, $transitions, kdCsrfTokenService, $log, $state, $q, $resource, localizerService) {
+    this.cookies_ = $cookies;
+    this.transitions_ = $transitions;
+    this.state_ = $state;
+    /** @private {!angular.$q.Promise} */
+    this.tokenPromise_ = kdCsrfTokenService.getTokenForAction('login');
+    this.q_ = $q;
+    this.resource_ = $resource;
+    this.jwtCookieName_ = 'kdToken';
+    this.skipLoginPageCookieName = 'skipLoginPage';
+    this.log_ = $log;
+    this.localizerService_ = localizerService;
+  }
+
+  setJWTCookie(token) {
+    this.cookies_.put(this.jwtCookieName_, token)
+  }
+
+  /**
+   *
+   * @param {!backendApi.LoginSpec} loginSpec
+   */
+  logIn(loginSpec) {
+    let deferred = this.q_.defer();
+
+    this.tokenPromise_.then(
+        token => {
+          let resource = this.resource_('api/v1/login', {}, {
+            save: {
+              method: 'POST',
+              headers: {
+                'X-CSRF-TOKEN': token,
```

Comment:
ah, too many vars named `token`. rename `tokenPromise_` to `csrfTokenPromise_`, and rename `token` here to `csrfToken` for clarity.